src/Features/TransientNotices.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * WooCommerce Transient Notices
+ * NOTE: DO NOT edit this file in WooCommerce core, this is generated from woocommerce-admin.
+ */
+
+namespace Automattic\WooCommerce\Admin\Features;
+
+use \Automattic\WooCommerce\Admin\Loader;
+
+/**
+ * Shows print shipping label banner on edit order page.
+ */
+class TransientNotices {
+
+	/**
+	 * Option name for the queue.
+	 */
+	const QUEUE_OPTION = 'woocommerce_admin_transient_notices_queue';
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		add_action( 'admin_enqueue_scripts', array( __CLASS__, 'enqueue_notices' ) );
+	}
+
+	/**
+	 * Add a notice to be shown.
+	 *
+	 * @param array $notice Notice.
+	 *    $notice = array(
+	 *      'id'      => (string) Unique ID for the notice. Required.
+	 *      'status'  => (string) info|error|success
+	 *      'content' => (string) Content to be shown for the notice. Required.
+	 *      'options' => (array) Array of options to be passed to the notice component.
+	 *       See https://developer.wordpress.org/block-editor/reference-guides/data/data-core-notices/#createNotice for available options.
+	 *    ).
+	 */
+	public static function add( $notice ) {
+		$queue = get_option( self::QUEUE_OPTION, array() );
```

Comment:
I pushed a commit showing this in action, but I'd prefer to roll it back.  I'm not as big a fan of doing this when methods contain more than a few arguments because:

1. It becomes difficult to use and remember the order of arguments.
```
TransientNotices::add(
    array(
        'user_id' => get_current_user_id(),
        'id'      => 'plugin-installer-' . str_replace( ',', '-', $plugins ),
        'status'  => 'success',
        'content' => $message,
    )
);
TransientNotices::add(
    'plugin-installer-' . str_replace( ',', '-', $plugins ),
    $message,
    array(),
    'success',
    get_current_user_id()
);
```
Using the latter I managed to make mistakes in both the readme and plugin installer since I mixed up the order of arguments.

2. Introducing new variables requires providing defaults for unused arguments and the consumer knowing more about the method.  This also reduces backwards compatibility.

If we're only concerned about enforcing required fields, we can add checks with the array style argument to enforce fields.

Reply:
+1 to rolling back this change - the associative array approach is more inline with [WP standards](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#self-explanatory-flag-values-for-function-arguments).

Reply:
@joshuatf sounds good 👍